src/rtc.rs

```diff
@@ -0,0 +1,180 @@
+        };
+
+        let freq = lse.freq().0;
+        let prl = freq - 1;
+        assert!(prl < 1 << 20);
+        result.perform_write(|s| {
+            s.regs.prlh.write(|w| unsafe { w.bits(prl >> 16) });
+            s.regs.prll.write(|w| unsafe { w.bits(prl as u16 as u32) });
+        });
+
+        result
+    }
+
+    /// Set the current rtc value to the specified amount of seconds
+    pub fn set_seconds(&mut self, seconds: u32) {
+        self.perform_write(|s| {
+            s.regs.cnth.write(|w| unsafe{w.bits(seconds >> 16)});
+        });
+        self.perform_write(|s| {
+            s.regs.cntl.write(|w| unsafe{w.bits(seconds)});
+        });
+    }
+
+    /**
+      Sets the time at which an alarm will be triggered
+
+      This also clears the alarm flag if it is set
+    */
+    pub fn set_alarm(&mut self, seconds: u32) {
+        // Set alarm time
+        // See section 18.3.5 for explanation
+        let alarm_value = seconds - 1;
+        self.perform_write(|s| {
+            s.regs.alrh.write(|w| unsafe{w.alrh().bits((alarm_value >> 16) as u16)});
+        });
+        self.perform_write(|s| {
+            s.regs.alrl.write(|w| unsafe{w.alrl().bits((alarm_value & 0x0000ffff) as u16)});
+        });
+
+        self.clear_alarm_flag();
+    }
+
+    /// Enables the RTCALARM interrupt
+    pub fn listen_alarm(&mut self) {
+        // Enable alarm interrupt
+        self.perform_write(|s| {
+            s.regs.crh.modify(|_, w| w.alrie().set_bit());
+        })
+    }
+
+    /// Disables the RTCALARM interrupt
+    pub fn unlisten_alarm(&mut self) {
+        // Disable alarm interrupt
+        self.perform_write(|s| {
+            s.regs.crh.modify(|_, w| w.alrie().clear_bit());
+        })
+    }
+
+    /// Reads the current time
+    pub fn read_seconds(&self) -> u32 {
+        // Wait for the APB1 interface to be ready
+        while self.regs.crl.read().rsf().bit() == false {}
+
+        ((self.regs.cnth.read().bits() << 16) as u32) + (self.regs.cntl.read().bits() as u32)
+    }
+
+    /// Enables the RTC second interrupt
+    pub fn listen_seconds(&mut self) {
+        self.perform_write(|s| {
+            s.regs.crh.modify(|_, w| w.secie().set_bit())
+        })
+    }
+
+    /// Disables the RTC second interrupt
+    pub fn unlisten_seconds(&mut self) {
+        self.perform_write(|s| {
+            s.regs.crh.modify(|_, w| w.secie().clear_bit())
+        })
+    }
+
+    /// Clears the RTC second interrupt flag
+    pub fn clear_second_flag(&mut self) {
+        self.perform_write(|s| {
+            s.regs.crl.modify(|_, w| w.secf().clear_bit())
+        })
+    }
+
+    /// Clears the RTC alarm interrupt flag
+    pub fn clear_alarm_flag(&mut self) {
+        self.perform_write(|s| {
+            s.regs.crl.modify(|_, w| w.alrf().clear_bit())
+        })
+    }
+
+    /**
+      Return `Ok(())` if the alarm flag is set, `Err(nb::WouldBlock)` otherwise.
+
+      **Note**: Does not clear the alarm flag
```

Comment:
Any reason of this? I'm OK with this, just curious.

Reply:
I thought that was the way it is done in the general purpose timers but I guess it's not. I'll make it clear the flag.